### DISCUSSION TITLE
Hardware and Setup Recommendations

### DISCUSSION THREAD

**Discussion by MoveQuick** [awawa-dev/HyperHDR#777] [Q&A]
Hello! 

Please forgive me if this is an extremely underinformed set of questions, I have been researching HyperHDR and Ambilight setups all day and am having trouble finding any concrete recommendations. I am hoping someone here can give me some suggestions and perhaps point me in the right direction.

I currently have a 55 inch LG C2 (not rooted) and if I pull the trigger on this setup will be getting an Apple tv 4k instead of using the apps on the TV. Also I plan to eventually buy a PS5 for gaming as well as likely use moonlight on the Apple TV for PC gaming.

As such, I am hoping to find a setup that can accommodate 4K, 120hz, HDR (both 10 and Dolby Vision) with minimal lag for gaming.

I've seen the recommendations for hardware here: https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html Utilizing a Raspberry pi 4 and the capture card they mention there and I am wondering if this would work well for the things I am trying to do. 

I have also seen that you can run HyperHDR through WebOS if you root your TV, Would this perhaps be the best option in this case? I've seen that it can stop you from viewing copyrighted content through this method, would this be a problem if I were to use an AppleTV with this setup? 

If not, is there some better options? (perhaps utilizing a splitter to have a direct hdmi cable to the TV and then a separate cable to the capture card to avoid latency?) I realize that it will be more expensive to buy better capture cards and such, but I am just wanting to know what would work best. I don't mind spending a few hundred dollars on this.

Again, my apologies if these are not the correct questions to be asking, just trying to wrap my head around this and potentially build a great setup to use this! Thanks in advance for any help!

**Comment by awawa-dev** [comment]
> As such, I am hoping to find a setup that can accommodate 4K, 120hz, HDR (both 10 and Dolby Vision) with minimal lag for gaming.

And propably VRR support? I haven't tested a similar setup for such requirements myself. This is a high bar for available devices. Dolby Vision will output LLDV if you choose the right splitter and this is the only protocol that can be captured by USB grabbers. HDCP similarly must be removed by the splitter otherwise the grabber will not capture the signal. Probably the new Ezcoo model https://www.amazon.de/Splitter-HDCP2-3-Vision-Scaler-Monitors/dp/B0C739QQ4Q (disclaimer: I haven't tested it) would be able to handle 4k120/VRR and force LLDV and remove HDCP for e.g. Elgato HD60x or similar grabber. But its price is a joke: over 2 x than previous older model. 

These are very expensive solutions, so you can start with webos if you feel up to rooting it (loss of warranty and in newer firmwares it is patched anyway). with webOS, all this should work except for playing protected content from the application ON the TV. HDCP via HDMI, from what I've heard before, should work. The quality and lag on webos depend on the model, the differences can be significant and, at least for now maybe there could be an some improvement soon, the colors quality may differ from what you can get with a video grabber.

And contrary to what you might expect, the USB grabber can deliver a lower lag signal than webos:
1) using USB grabber:
HDMI -> USB grabber -> USB -> HyperHDR usually with ~50ms with tone mapping
2) using webos capturing:
HDMI -> TV -> signal processing, cinema mode in some models may take even over 200ms but gaming mode should be much lower => webos capturing => encapsulation to flatbuffer => HyperHDR => unpacking flatbuffer and applying tone mapping

With the first solution, sometimes you encounter situations where the LEDs are faster than the TV, although it's really hard to notice and eg. you have to record a video (TV & LEDs) using a high frame rate.

**Comment by MoveQuick** [comment]
Thank you so much for the reply!

So just wanting to make sure I understand how this would be connected.

I would have the following:

Apple TV 4k > Ezcoo splitter > 4k signal > TV/ 1080p signal > HD60X 
From the capture card I would run to a raspberry pi (4 model b or would a 5 be worth it? I saw that it doesn't seem to be fully supported yet) and follow the rest of the previously mentioned guide to setup the light strip.

Is that correct? Thank you again for the help. While I'm pretty good with tech, this is all relatively new to me, so I really appreciate it.

**Reply by awawa-dev** [reply]
It looks OK, but as I mentioned, the HDMI splitter is still an unknown factor as I don't have it nor have I tested the one you need because it's out of my testing budget range for now. Not sure which LED driver are you planning to use...my short and possibly biased opinion about RPi5: https://github.com/awawa-dev/HyperHDR/discussions/661

**Reply by LAKSHITgoel** [reply]
Is there any to run HyperHDR in AppleTV4k (tvOS) itself and just use an LED Strip and ESP32 or something? I don't want to use HDMI capture card and setups because I am using HDMI erc for DolbyAtmos sound setup and I do not want to compromise on the picture quality and sound at all.
